syntax = "proto3";

package dataflow;

import "google/protobuf/empty.proto";

// Record message for individual record processing
message Record {
  string row_data = 1;
}

// Define RecordBatch message for batch processing
message RecordBatch {
  repeated Record records = 1;
}

service DataService {
  // Send individual record
  rpc SendRecord(Record) returns (google.protobuf.Empty) {}
  
  // Send batch of records - make sure method name matches implementation
  rpc SendRecordBatch(RecordBatch) returns (google.protobuf.Empty) {}
  
  // Signal end of stream
  rpc EndStream(google.protobuf.Empty) returns (google.protobuf.Empty) {}
}